### ISSUE TITLE
NPE in SonarBBPluginConfig.validate

### ISSUE THREAD

**Issue by psnitil** [mibexsoftware/sonar-bitbucket-plugin#32]
Hi, I'm trying to configure this plugin (v1.1.5) with our SonarCuber server (v6.1) and I'm getting NullPointerException during scans. Am I missing something in the configuration or could be something wrong with sonar cube/plugin instalation? 

```
[Sonar] 15:35:35.638 ERROR: Error during SonarQube Scanner execution
[Sonar] java.lang.NullPointerException
[Sonar] 	at ch.mibex.bitbucket.sonar.SonarBBPluginConfig.validate(SonarBBPluginConfig.scala:63)
[Sonar] 	at ch.mibex.bitbucket.sonar.cache.InputFileCacheSensor.shouldExecuteOnProject(InputFileCacheSensor.scala:17)
[Sonar] 	at org.sonar.scanner.bootstrap.BatchExtensionDictionnary.shouldKeep(BatchExtensionDictionnary.java:260)
[Sonar] 	at org.sonar.scanner.bootstrap.BatchExtensionDictionnary.getFilteredExtensions(BatchExtensionDictionnary.java:101)
[Sonar] 	at org.sonar.scanner.bootstrap.BatchExtensionDictionnary.select(BatchExtensionDictionnary.java:74)
[Sonar] 	at org.sonar.scanner.phases.SensorsExecutor.execute(SensorsExecutor.java:45)
[Sonar] 	at org.sonar.scanner.phases.AbstractPhaseExecutor.execute(AbstractPhaseExecutor.java:78)
[Sonar] 	at org.sonar.scanner.scan.ModuleScanContainer.doAfterStart(ModuleScanContainer.java:182)
[Sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
[Sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
[Sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.scan(ProjectScanContainer.java:247)
[Sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.scanRecursively(ProjectScanContainer.java:242)
[Sonar] 	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:232)
[Sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
[Sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
[Sonar] 	at org.sonar.scanner.task.ScanTask.execute(ScanTask.java:47)
[Sonar] 	at org.sonar.scanner.task.TaskContainer.doAfterStart(TaskContainer.java:86)
[Sonar] 	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:142)
[Sonar] 	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:127)
[Sonar] 	at org.sonar.scanner.bootstrap.GlobalContainer.executeTask(GlobalContainer.java:115)
[Sonar] 	at org.sonar.batch.bootstrapper.Batch.executeTask(Batch.java:118)
[Sonar] 	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:62)
[Sonar] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[Sonar] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[Sonar] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[Sonar] 	at java.lang.reflect.Method.invoke(Method.java:498)
[Sonar] 	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
[Sonar] 	at com.sun.proxy.$Proxy0.execute(Unknown Source)
[Sonar] 	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:233)
[Sonar] 	at org.sonarsource.scanner.api.EmbeddedScanner.runAnalysis(EmbeddedScanner.java:151)
[Sonar] 	at org.sonarsource.scanner.cli.Main.runAnalysis(Main.java:110)
[Sonar] 	at org.sonarsource.scanner.cli.Main.execute(Main.java:74)
[Sonar] 	at `org.sonarsource.scanner.cli.Main.main(Main.java:61)
```

**Comment by mrueegg** [comment]
Hi,

How do you trigger the plug-in? With Maven, SonarRunner or something else? The command line you use to trigger it would be of much help in debugging this problem.

**Comment by imunic** [comment]
Hi,

Got the same error today when trying both gradle and sonar scanner, for some reason the `server.getVersion` call returns `null`, the other parameters are correctly passed and populated (startup time and url, if it helps the instance of the server object is `DefaultServer`). 

Using SonarQube 6.1 also. I can see the version correctly reported via the api or in the log of sonar scanner.

Tried rebuilding the plugin with the mentioned server version check commented out and after that it worked as expected.

**Comment by mrueegg** [comment]
Hi,

I have no idea why `server.getVersion` returns `null`, as this is a core property that should always be available. But I will make the version check optional in that case and only fail in case the version is == 5.1 which is not supported by the plug-in (see SONAR-6398).

**Comment by lopesdasilva** [comment]
I have this problem as well. 
I'm using Sonar version 6.1 
And the latest version jar of the plugin.

**Comment by mrueegg** [comment]
I've released the new version 1.1.6 with a bug fix. Thanks for bringing this up!
